src/base/net/stratum/SelfSelectClient.cpp

```diff
@@ -235,6 +240,51 @@ void xmrig::SelfSelectClient::submitBlockTemplate(rapidjson::Value &result)
     });
 }
 
+int64_t xmrig::SelfSelectClient::submit(const JobResult& result)
+{
+    if (m_submit_to_origin) {
+        submitOriginDaemon(result);
+    }
+    return m_client->submit(result);
+}
+
+void xmrig::SelfSelectClient::submitOriginDaemon(const JobResult& result)
+{
+    if (result.diff == 0 || m_targetdiff == 0) {
+        return;
+    }
+    
+    if (result.actualDiff() < m_targetdiff) {
+        m_origin_not_submitted++;
+        LOG_INFO("%s " RED_BOLD("not submitted to origin daemon, difficulty too low") " (%" PRId64 "/%" PRId64 ") "
+            BLACK_BOLD(" diff ") BLACK_BOLD("%" PRIu64) BLACK_BOLD(" vs. ") BLACK_BOLD("%" PRIu64),
+            Tags::origin(), m_origin_submitted, m_origin_not_submitted, m_targetdiff, result.actualDiff());
+        return;
+    }
+    char *data = m_blocktemplate.data();
+    Cvt::toHex(data + 78, 8, reinterpret_cast<const uint8_t*>(&result.nonce), 4);
+
+    using namespace rapidjson;
+    Document doc(kObjectType);
+
+    Value params(kArrayType);
+    params.PushBack(m_blocktemplate.toJSON(), doc.GetAllocator());
+
+    JsonRequest::create(doc, m_sequence, "submitblock", params);
+    m_results[m_sequence] = SubmitResult(m_sequence, result.diff, result.actualDiff(), 0, result.backend);
+
+    FetchRequest req(HTTP_POST, pool().daemon().host(), pool().daemon().port(), "/json_rpc", doc, pool().daemon().isTLS(), isQuiet());
+    fetch(tag(), std::move(req), m_httpListener);
+    
+    // Ensure that the origin daemon request the latest blockchain height in case the submission was successful and it changed
+    FetchRequest req_height(HTTP_GET, pool().daemon().host(), pool().daemon().port(), "/getheight", pool().daemon().isTLS(), isQuiet());
```

Comment:
How is this result handled?

Reply:
In our merge mining proxy implementation, i.e. "origin daemon", a `"/getheight"` call from XMrig to the proxy results in the proxy requesting height from `monerod` and the base node. When it senses a height change in any of the Tari or Monero blockchains, it is then able to push a new block template to XMRig.

With XMRig in `daemon` mode, `"/getheight"` to the proxy runs in a timer job approximately every second inside `DaemonClient`.

With this `self-select` + `submit-to-origin` mode, `"/getheight"` to the proxy runs after each block submission to the proxy, which will result in the same height change sensing behaviour as described above.